### DISCUSSION TITLE
Duplicate Matrix Block

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8123] [Ideas]
> Created by: Julian Rutherford (justjoolz@gmail.com) on 2015/04/19 11:42:02 +0000
> Votes at time of UserVoice import: 26

-----------------------------------

An option to 'duplicate this block', on the little cog for matrix blocks would be great.

**Comment by angrybrad** [comment]
> Posted by Anonymous (nate@blendimc.com) on 2017/01/18 21:36:50 +0000
-----------------------------------

This would save so much time.

**Comment by angrybrad** [comment]
> Posted by Jason Morehead (jasonmorehead@gmail.com) on 2015/03/20 15:23:53 +0000
-----------------------------------

I&#39;ve got a situation right now where duplicating matrix blocks would be really handy.

**Comment by angrybrad** [comment]
> Posted by Hermien (hello@hermienscholten.nl) on 2016/07/28 13:52:57 +0000
-----------------------------------

this would be so helpful :o +1

**Comment by angrybrad** [comment]
> Posted by Bert Oost (b.oost@vrieling.nl) on 2016/06/07 12:36:07 +0000
-----------------------------------

This would be so useful! +10 :-)

**Comment by Ottergoose** [comment]
*= 100

**Comment by proimage** [comment]
I'd love this. Specifically, I'd love to have a Redactor button to duplicate the matrix block while splitting the Redactor content before and after the cursor position between the original and new blocks. That just seems like one of those smooth features Craft would have, y'know?

**Comment by brandonkelly** [comment]
@proimage Good idea! Can you post as a new FR at https://github.com/craftcms/redactor/issues please?

**Comment by proimage** [comment]
@brandonkelly ~Will do~ [Done](https://github.com/craftcms/redactor/issues/1), but it would also be nice to be able to split plain textareas in the same fashion, for those people who use plaintext markup (eg. Markdown) for writing longform content. :)

**Comment by andrewhawkes** [comment]
Are we any closer to getting this as a feature?

**Comment by brandonkelly** [comment]
@andrewhawkes No progress has been made on it yet, but in the meantime you can use Verbb’s cleverly-named [Smith](https://plugins.craftcms.com/smith) plugin.

**Comment by andrewhawkes** [comment]
@brandonkelly okay thanks, I'll check that out!
